### PR TITLE
Event page adjusts to key and transaction status

### PR DESCRIPTION
**Pull request by benwerd**
# Description

The event page now reacts properly to key status. It will show as confirmed if you've bought a key, confirming if this is in progress, etc.

**Please only evaluate latest commit.** There will be one more cascading commit tonight, to produce the QR code that sits on the event page itself. Obviously this should be rebased before it's merged in, with the QR code component coming first. [There is a configuration change here, which will disappear when rebased. Please ignore.]

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: tickets/.eslintrc.js [hunk 1]
```diff
@@ -14,6 +14,14 @@ module.exports = {
   rules: {
     'react/prefer-stateless-function': [2],
     'react/forbid-prop-types': 2,
+    '@typescript-eslint/no-unused-vars': [
```
Comment (julien51) [inline tickets/.eslintrc.js]: Any reason why you add this eslint rule? Maybe better on another PR?

File: tickets/src/components/content/purchase/PayButton.js [hunk 2]
```diff
@@ -24,9 +26,13 @@ export const PayButton = ({ transaction, purchaseKey, config }) => {
         {requiredConfirmations})
       </PayInfo>
     )
-  } else if (transaction && transaction.status === 'mined') {
+  } else if (keyStatus === KeyStatus.CONFIRMING) {
+    // Key is being confirmed but we don't have transaction
```
Comment (julien51) [inline tickets/src/components/content/purchase/PayButton.js]: Hum. How is that possible? The transaction should come before the key?

File: tickets/.eslintrc.js (hunk 1 above)
Comment (cnasc) [reply tickets/.eslintrc.js]: This rule should already be covered by our eslint config (and `tsc` also catches it)

File: tickets/src/components/content/purchase/PayButton.js (hunk 2 above)
Comment (cellog) [reply tickets/src/components/content/purchase/PayButton.js]: I think you should copy the logic from `paywall/src/components/lock/Lock.js` here

**Pull request merged**